precompiles/erc20factory/types.go

```diff
@@ -0,0 +1,101 @@
+package erc20factory
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	cmn "github.com/cosmos/evm/precompiles/common"
+)
+
+// EventCreate defines the event data for the ERC20 Factory Create event.
+type EventCreate struct {
+	TokenAddress    common.Address
+	TokenPairType   uint8
+	Salt            [32]uint8
+	Name            string
+	Symbol          string
+	Decimals        uint8
+	Minter          common.Address
+	PremintedSupply *big.Int
+}
+
+// ParseCreateArgs parses the arguments from the create method and returns
+// the token type, salt, name, symbol, decimals, minter, and preminted supply.
+func ParseCreateArgs(args []interface{}) (tokenType uint8, salt [32]uint8, name string, symbol string, decimals uint8, minter common.Address, premintedSupply *big.Int, err error) {
+	if len(args) != 7 {
+		return uint8(0), [32]uint8{}, "", "", uint8(0), common.Address{}, nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 7, len(args))
+	}
+
+	tokenType, ok := args[0].(uint8)
+	if !ok {
+		return uint8(0), [32]uint8{}, "", "", uint8(0), common.Address{}, nil, fmt.Errorf("invalid tokenType")
+	}
+
+	salt, ok = args[1].([32]uint8)
+	if !ok {
+		return uint8(0), [32]uint8{}, "", "", uint8(0), common.Address{}, nil, fmt.Errorf("invalid salt")
+	}
+
+	name, ok = args[2].(string)
+	if !ok || len(name) < 3 || len(name) > 128 {
+		return uint8(0), [32]uint8{}, "", "", uint8(0), common.Address{}, nil, fmt.Errorf("invalid name")
+	}
+
+	symbol, ok = args[3].(string)
+	if !ok || len(symbol) < 3 || len(symbol) > 16 {
```

Comment:
We should allow the symbol to be 1 character. There are cases like this in production already, like Sonic's $S.